modules/programs/offlineimap.nix

```diff
@@ -0,0 +1,174 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+with import ../lib/dag.nix { inherit lib; };
+with import ../lib/mail.nix { inherit lib config; };
+
+let
+  cfg = config.programs.offlineimap;
+
+  accounts =
+    filter (a: a.notmuch.enable)
+    (attrValues config.accounts.email.accounts);
+
+  # generates a script to fetch only a specific account
+  genOfflineImapScript = {name, ...} @ account:
+    pkgs.writeShellScriptBin "offlineimap-${name}" ''
+      ${pkgs.offlineimap}/bin/offlineimap -a${account.name} -c "$XDG_CONFIG_HOME/offlineimap/config" $@
```

Comment:
I don't think you need the `-c` flag here since, from what I can tell, offlineimap will look for a config in `$XDG_CONFIG_HOME/offlineimap/config` by default: https://github.com/OfflineIMAP/offlineimap/commit/5150de5